sdccc/src/test/java/com/draeger/medical/sdccc/tests/util/MdibHistorianTest.java

```diff
@@ -855,54 +865,206 @@ void testUniqueEpisodicReportBasedHistoryGoodDoNotFilterWhenReportTypeDiffers()
+        report.setSequenceId(sequenceId);
+
+        testApplyReportOnStorage(numericMdibVersion, sequenceId, report);
+    }
 
-        final var firstMod =
-                (EpisodicContextReport) contextReport.getBody().getAny().get(0);
-        firstMod.getReportPart().get(0).getContextState().add(operator1);
-        firstMod.getReportPart().get(0).getContextState().add(operator2);
+    @Test
+    void testUniqueEpisodicReportBasedHistoryUntilTimestampRegressionWithOperationInvokedReport()
+            throws ReportProcessingException, JAXBException, PreprocessingException, IOException {
+
+        final BigInteger numericMdibVersion = BigInteger.ZERO;
+        final String sequenceId = "abc";
+
+        final var messageObjectFactory = new ObjectFactory();
+        final var participantObjectFactory = new com.draeger.medical.biceps.model.participant.ObjectFactory();
+
+        final var report = messageObjectFactory.createOperationInvokedReport();
+        final var part = messageObjectFactory.createOperationInvokedReportReportPart();
+        final var invInfo = messageObjectFactory.createInvocationInfo();
+        invInfo.setInvocationError(com.draeger.medical.biceps.model.message.InvocationError.OTH);
+        invInfo.setInvocationState(com.draeger.medical.biceps.model.message.InvocationState.FAIL);
+        invInfo.setTransactionId(123);
+        part.setInvocationInfo(invInfo);
+        part.setOperationTarget("opTarget");
+        part.setOperationHandleRef("opHandle");
+
+        final var invSrc = participantObjectFactory.createInstanceIdentifier();
+        final var type = participantObjectFactory.createCodedValue();
+        type.setCode("33");
+        type.setCodingSystem("JOCS");
+        type.setCodingSystemVersion("2.0");
+        invSrc.setType(type);
+        invSrc.setRootName("rootName");
+        invSrc.setExtensionName("extName");
+        part.setInvocationSource(invSrc);
+        part.setSourceMds("sourceMds");
+        report.getReportPart().add(part);
+        report.setMdibVersion(numericMdibVersion);
+        report.setSequenceId(sequenceId);
+        final Envelope soapMessage =
+                messageBuilder.createSoapMessageWithBody(ActionConstants.ACTION_OPERATION_INVOKED_REPORT, report);
+
+        testUniqueEpisodicReportBasedHistoryUntilTimestamp(sequenceId, soapMessage);
+    }
 
-        final var contextReport2 = buildEpisodicContextReport(
-                MdibBuilder.DEFAULT_SEQUENCE_ID, contextReportMdibVersion, contextReportStateVersion);
-        final var secondMod =
-                (EpisodicContextReport) contextReport2.getBody().getAny().get(0);
-        secondMod.getReportPart().get(0).getContextState().add(operator1);
+    @Test
+    void testEpisodicReportBasedHistoryRegressionWithOperationInvokedReport()
+            throws ReportProcessingException, JAXBException, PreprocessingException, IOException {
+
+        // given
+        final var messageObjectFactory = new ObjectFactory();
```

Comment:
You do not need to instantiate the factory here, you can simply use the MessageBuilder.

Reply:
Indeed, Done.